Tests/IntegrationTests/PSQLIntegrationTests.swift

```diff
@@ -360,4 +360,28 @@ final class IntegrationTests: XCTestCase {
         }
     }
     
+    func testConnectionClosureMidQueryDoesNotHang() async throws {
+        let badQuery: PostgresQuery = """
+        SELECT * FROM non_existent_table
+        """
+
+        _ = try await withThrowingTaskGroup(
+            of: Void.self
+        ) { taskGroup in
+            for _ in (0 ..< 1_000) {
+                taskGroup.addTask {
+                    let conn = try await PostgresConnection.test(
+                        on: NIOSingletons.posixEventLoopGroup.next()
```

Comment:
@fabianfett not sure if you're fine with me just using the singleton ELG. It appears PostgresNIO already requires a high-enough SwiftNIO version.